### ISSUE TITLE
Transparency support for PNG

### ISSUE THREAD

**Issue by tryadelion** [stefanhaustein/TerminalImageViewer#77]
there seems to be no support for alpha channel on PNG's / Gifs, is that a possibility? atleast on fully transparent rgba(0,0,0,0) pixels.

**Comment by aaronliu0130** [comment]
I don't think that is as this project uses unicode half-block characters, and I don't think there is a transparent half block character. If we use unicode invisible characters, we can't really promise on the spacing.

**Comment by tryadelion** [comment]
how about U+2591 for 0-alpha? it oulndt look solid, atleast

**Comment by stefanhaustein** [comment]
We can't access / preserve the terminal background. We could add support for a user-specified "background" color though.... Perhaps default to the common gray/gray checkerboard pattern?

**Comment by aaronliu0130** [comment]
> how about U+2591 for 0-alpha? it wouldnt look solid, atleast

Yes, that Unicode character sure doesn't work well on my terminal emulator. See the screenshot below.
![Ignore the cow.](https://user-images.githubusercontent.com/11874211/83405754-d021ea80-a43f-11ea-821e-4c3f2b332799.png)

**Comment by tryadelion** [comment]
@stefanhaustein maybe sth like this? https://stackoverflow.com/questions/43931304/is-it-possible-to-determine-the-background-color-of-the-terminal-in-a-linux-mach

**Comment by stefanhaustein** [comment]
Not sure how accessible the shell variable would be from cross-platform c++... if not, an explicit command line parameter would still help to pass it over... So I think it makes sense to start with that and then take it from there....

**Comment by aaronliu0130** [comment]
Well, usage of `system()` is considered to be not good, but since we don't have any other options lets just take it from here. I'm not sure if that shell script is cross-terminal though...

**Comment by tryadelion** [comment]
I can test on some terminals mac and debian, mint and a couple more if necessary
